test/integration/node-specific/mongo_client.test.ts

```diff
@@ -722,6 +722,22 @@ describe('class MongoClient', function () {
         expect(client.s.sessionPool.sessions).to.have.lengthOf(1);
       });
     });
+
+    context('concurrent calls', () => {
+      context('when two concurrent calls to close() occur', () => {
```

Comment:
I can't get either of these tests to fail when I revert the changes you made to the MongoClient.  Are you sure these tests are working as expected?

Reply:
Good catch, I'll add in more resources for client.close() to clean-up, by calling connect / setting a poolSize, and see if we can get the tests to fail without the changes.